### PR TITLE
Adds framework for the documentation of analyzed tools

### PR DESCRIPTION
**Pull request by nabialek-arianelabs**
Adds framework for research documentation of EVM analytic tools: 
* creates directories template
*  adds boilerplate files: tested smart contracts, execution reports and exceptions
* creates README file

### CODE REVIEW

**Comment by acuarica** [comment]
Hi @nabialek-arianelabs, thanks for sending this. Is there any particular reason to include specific analysis files, `maian-analysis` and `slither-analysis`, in this PR? Does it make sense to include these in their respective PRs? This way the other PRs would be self-contained.

**Review by acuarica — COMMENTED** [review]
Just went through the `README`, added a couple of details.

I haven't gone through the rest as per https://github.com/hashgraph/hedera-smart-contracts/pull/726#issuecomment-2099056259.

File: tools/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+# EVM Analytic tools study
+### General Information:
+This report aims to analyze potential tools that will aid in the development, porting, and security assessment
```
[the remark this answers is not in the document]
Comment (arianejasuwienas) [reply tools/README.md:3]: @acuarica This sentence is correct. The focus of these reports is the analysis of the tools.

File: tools/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+of Smart Contracts deployed on the Hedera network. Key aspects evaluated during this research include:
+- Ease of use and compatibility with Hedera.
+- Ability to identify optimizations and potential errors or bugs specific to Smart Contracts on the Hedera.
+- Issues with compatibility for contracts ported from Ethereum or other networks to the Hedera.
+### Prerequisites
+Tools were tested in the MacOS and Kubuntu Linux environments as well as on Docker containers. Test setup involves:
+- Installing the Solidity compiler package.
+- Cloning the Hedera JSON RPC repository: `git clone -b main --single-branch  https://github.com/hashgraph/hedera-json-rpc-relay.git`.
+- Installing dependencies, and building the project: `npm install`, `npm run setup`, from the project directory.
+- Starting the project: `npm start`.
```
[the remark this answers is not in the document]
Comment (arianejasuwienas) [reply tools/README.md]: @acuarica Our analysis included investigating the possibility of replacing their APIs with Hedera Json Rpc in MAIAN and Manticore tools.

But I guess you are right - it is not crucial for the final report, so I've removed it.

**Comment by acuarica** [comment]
Hi @arianejasuwienas, we have discussed with the team the main points of these PRs. From its contents, it is not entirely clear what the purpose of these PRs is. However, from your comments it seems the main goal is to evaluate the suitability of these tools in a Hedera project. If that's the case, could you rewrite the docs to make that explicit?

There is no need to include full report and logs output, at most, some sample report output might be embedded in the docs. All the docs may be placed inside a folder `tools/analysis` (leaving room for other kind of tools in the future). Instructions and config files, _e.g._, for docker, on how to run these tools might be included as well.

Without the report and logs output the PRs might be just a few files. If that's the case, it might be easier to merge them into a single PR (like the original one posted in the relay https://github.com/hashgraph/hedera-json-rpc-relay/pull/2439).

The references for Etherscan API keys don't apply for Hedera, those should be removed.

Happy to discuss this further is something is not clear.

_team in cc_ @ebadiere @AlfredoG87 @quiet-node

**Comment by arianejasuwienas** [comment]
@acuarica 

Test reports are a critical component of our research. Removing them entirely from the contexts in which we [draw conclusions](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R251) from performed tests would be challenging.

Perhaps, in such cases, we could [maintain these reports in a separate repository](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R218) and reference them in this specific PR.

Omitting Etherscan API information entirely means we lose essential details: Slither's ability to access resources directly from the blockchain is not compatible with Hashscan. My proposal is to condense this information but still retain it in the [report](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R97).

**Comment by acuarica** [comment]
Hey @arianejasuwienas,

> Test reports are a critical component of our research. Removing them entirely from the contexts in which we [draw conclusions](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R251) from performed tests would be challenging.

I might be missing something here, but why it would be challenging? If the focus is on the Smart Contracts themselves, I agree, but this is not the case. Some example report might be worth including, but not all of them.

> Perhaps, in such cases, we could [maintain these reports in a separate repository](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R218) and reference them in this specific PR.

Sure, that might be a potential solution. However, having these reports in a separate repo begs the question of whether to move the whole evaluation into a separate repo.

> Omitting Etherscan API information entirely means we lose essential details: Slither's ability to access resources directly from the blockchain is not compatible with Hashscan. My proposal is to condense this information but still retain it in the [report](https://github.com/hashgraph/hedera-smart-contracts/pull/738/files#diff-72e679330dba9af7956416802a2f76b4c804e336b2ae4e0a5a2083550a6d6280R97).

One clarification first: Etherscan is both a block explorer and a verification service. Hashscan is only a block explorer. The verification service that we use is based on Sourcify and is provided by a different service.[*]

Here what's important is that `slither-read-storage` does not support Hedera networks. `slither-read-storage` seems to only support Etherscan-like services. I don't think the problem comes from `eth_getStorageAt`, I think the problem comes from not having the source code given the error `ERROR:SlitherSolcParsing:crytic-compile returned an empty AST. If you are trying to analyze a contract from etherscan or similar make sure it has source code available.`. From the `slither-read-storage` docs https://github.com/crytic/slither/blob/master/slither/tools/read_storage/README.md

> ## Limitations
> - Requires source code.

---

[*] Hashscan could in principle provide a Etherscan-like API given that it has the block information and can act as a facade for the verification service. This might be worth exploring in the future.

**Comment by arianejasuwienas** [comment]
Replaced with: https://github.com/hashgraph/hedera-smart-contracts/pull/740

**Pull request closed without merging**